### PR TITLE
Feature Flag: Allow to load the plugin as external

### PR DESCRIPTION
**Pull request by andresmgot**
**What is this feature?**

Allow to skip loading a core plugin so the plugin ID is "free" for an external plugin to use it.

To try it:

 - Make the new plugin folder available (as an external plugin). For example, creating a link: `ln -s ../../public/plugins/testdata data/plugins/tesdata`
 - Add the following config:

```
[plugins]
skip_core_plugins = testdata

[feature_toggles]
decoupleCorePlugins = true
```

**Which issue(s) does this PR fix?**:

Ref #68973

**Special notes for your reviewer:**

Please check that:
- [X] It works as expected from a user's perspective.
- [X] If this is a pre-GA feature, it is behind a feature toggle.
- [X] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/plugins/manager/loader/finder/local.go
Comment (xnyo) [inline pkg/plugins/manager/loader/finder/local.go:88]: Nit: Probably not a big deal now because this is under a feature flag, but I feel like this should be `Debug` level once the feature flag is removed :)

**Review by xnyo — APPROVED** [review]
LGTM! I tested it and it works great. Just one comment but feel free to ignore :wink:

**Review by wbrowne — COMMENTED** [review]
Nice work!

File: pkg/plugins/manager/loader/finder/local.go
Comment (andresmgot) [reply pkg/plugins/manager/loader/finder/local.go:83]: :sweat_smile: should I wait for your PR to be merged before I merge this?

Comment (wbrowne) [reply pkg/plugins/manager/loader/finder/local.go:83]: 😂 I can poke for some more reviews, but also don't want to block you!

File: pkg/plugins/manager/loader/finder/local.go
Comment (andresmgot) [reply pkg/plugins/manager/loader/finder/local.go:85]: this is ideally a filter (in the pipeline scenario, as you mentioned). My idea was to avoid any possible processing if the plugin is meant to be skipped but in any case I can move it in the meantime.

File: pkg/cmd/grafana-cli/services/services.go
Comment (andresmgot) [reply pkg/cmd/grafana-cli/services/services.go:81]: keeping this out

**Review by wbrowne — APPROVED** [review]
Nice one 🚀

**Pull request merged**
